src/server/server_test.go

```diff
@@ -0,0 +1,86 @@
+package server
+
+import (
+	"context"
+	"fenix/src/utils"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+func StartServer() (*utils.WaitGroupCounter, *http.Server, *ServerHub) {
+	wg := utils.NewWaitGroupCounter()
+	srv, hub := Serve("localhost:8080", wg)
+	wg.Wait()
+	return wg, srv, hub
+}
+
+func ConnectToServer(t *testing.T) *websocket.Conn {
```

Comment:
Test makes no assertion. Could maybe assert that the error is not nil.

Reply:
This is just the setup for tests.  Tests are shown by `TestXxx`

Go does not have `assert`.  In testing, you use `if <cond> ... t.Fatalf("Uh oh")`

You may see the following code:
```
	if err != nil {
		t.Fatalf("Error connecting to localhost:8080: %v", err)
	}
```

I assert that err is not nil and error if it is.